File: LeetCode/Python/1790_Check_if_One_String_Swap_Can_Make_Strings_Equal.py
# 1790. Check if One String Swap Can Make Strings Equal
# https://leetcode.com/problems/check-if-one-string-swap-can-make-strings-equal/


class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        char = 0
        for i, j in zip(s1, s2):
            if i != j:
                char += 1
        return s1 == s2 or sorted(s1) == sorted(s2) and char == 2

File: LeetCode/JavaScript/1790-Check-if-One-String-Swap-Can-Make-Strings-Equal.js
// 1790. Check if One String Swap Can Make Strings Equal
// https://leetcode.com/problems/check-if-one-string-swap-can-make-strings-equal/

/**
 * @param {string} s1
 * @param {string} s2
 * @return {boolean}
 */

var areAlmostEqual = function (s1, s2) {
  let chars = [],
    count = 0;

  for (let i = 0; i < s1.length; i++) {
    if (s1[i] !== s2[i]) {
      count++;
      chars.push(s1[i]);
      chars.push(s2[i]);
    }
    if (count > 2) return false;
  }

  let text1 = chars.map((v, i) => (i % 2 === 0 ? v : '')).join('');
  let text2 = chars
    .map((v, i) => (i % 2 !== 0 ? v : ''))
    .reverse()
    .join('');
  return text1 === text2 ? true : false;
};
